File: rtix/core/tests/test_logging.cpp
// Copyright (c) 2025 Velodex Robotics, Inc and RTIX Developers.
// Licensed under Apache-2.0. http://www.apache.org/licenses/LICENSE-2.0

#include <gtest/gtest.h>
#include <spdlog/spdlog.h>
#include "rtix/core/logging.h"

TEST(Core, Logging) {
  rtix::core::setupDefaultLogging("./cpp_test_log.log");

  SPDLOG_DEBUG("Debug");
  SPDLOG_INFO("Info");
  SPDLOG_WARN("Warning");
  SPDLOG_ERROR("Error");
}

File: rtix/core/tests/test_logging.py
# Copyright (c) 2025 Velodex Robotics, Inc and RTIX Developers.
# Licensed under Apache-2.0. http://www.apache.org/licenses/LICENSE-2.0

import logging
from rtix.core.logging import setupDefaultLogging


def test_logging():
    setupDefaultLogging("./py_test_log.log")

    logging.debug("Debug")
    logging.info("Info")
    logging.warning("Warning")
    logging.error("Error")
